cowprotocol/services
# Prepare domain submission config

# Description
Since settlement encoding is a crucial and hairy part of our logic we don't want to simply switch from the battle-tested existing system to a new implementation but rather make this configurable so we can roll it out per network/environment and have an easy lever to revert in case something goes wrong.

# Changes
- [x] Introduce new domain/boundary settlement encoding config type
- [x] Prepare conditional logic (with a TODO for the domain logic for now)

## How to test

cargo run --bin driver -- --ethrpc http://localhost:8545 --config crates/driver/example.toml

- See Boundary is used by default
- Add encoding = "Domain" to the top of the yaml file, see we are now using Domain encoding

## Related Issues

Part of #2215
